src/decisionengine/framework/tests/test_query_tool_server.py

```diff
@@ -59,56 +55,47 @@ def test_query_tool_csv(deserver):
         "5,value3,Test,test_channel,1,2\n"
-        "6,value1,0.1,test_channel,1,3\n"
-        "7,value2,2,test_channel,1,3\n"
-        "8,value3,Test,test_channel,1,3\n"
-        "\n"
     )
 
 @pytest.mark.usefixtures("deserver")
 def test_query_tool_json(deserver):
     # Test json output
     deserver.de_client_run_cli('--block-while', 'BOOT'),
     output = deserver.de_query_tool_run_cli('foo', '--format=json')
-    assert re.match(
-        r'Product foo:  Found in channel test_channel\n'
-        r'{\n'
-        r'\s*"key1":\s*{\n'
-        r'\s*"0":\s*"value1",\n\s*"1":\s*"value2",\n\s*"2":\s*"value3",\n'
-        r'\s*"3":\s*"value1",\n\s*"4":\s*"value2",\n\s*"5":\s*"value3",\n'
-        r'\s*"6":\s*"value1",\n\s*"7":\s*"value2",\n\s*"8":\s*"value3"\n'
-        r'\s*},\n'
-        r'\s*"key2":\s*{\n'
-        r'\s*"0":\s*0.1,\n\s*"1":\s*2,\n\s*"2":\s*"Test",\n'
-        r'\s*"3":\s*0.1,\n\s*"4":\s*2,\n\s*"5":\s*"Test",\n'
-        r'\s*"6":\s*0.1,\n\s*"7":\s*2,\n\s*"8":\s*"Test"\n'
-        r'\s*},\n'
-        r'\s*"channel":\s*{\n'
-        r'\s*"0":\s*"test_channel",\n\s*"1":\s*"test_channel",\n\s*"2":\s*"test_channel",\n'
-        r'\s*"3":\s*"test_channel",\n\s*"4":\s*"test_channel",\n\s*"5":\s*"test_channel",\n'
-        r'\s*"6":\s*"test_channel",\n\s*"7":\s*"test_channel",\n\s*"8":\s*"test_channel"\n'
-        r'\s*},\n'
-        r'\s*"taskmanager_id":\s*{\n'
-        r'\s*"0":\s*1,\n\s*"1":\s*1,\n\s*"2":\s*1,\n'
-        r'\s*"3":\s*1,\n\s*"4":\s*1,\n\s*"5":\s*1,\n'
-        r'\s*"6":\s*1,\n\s*"7":\s*1,\n\s*"8":\s*1\n'
-        r'\s*},\n'
-        r'\s*"generation_id":\s*{\n'
-        r'\s*"0":\s*1,\n\s*"1":\s*1,\n\s*"2":\s*1,\n'
-        r'\s*"3":\s*2,\n\s*"4":\s*2,\n\s*"5":\s*2,\n'
-        r'\s*"6":\s*3,\n\s*"7":\s*3,\n\s*"8":\s*3\n'
-        r'\s*}\n'
-        r'}\n',
-        output
-    )
+    assert output.startswith('Product foo:  Found in channel test_channel\n')
+
+    as_json = json.loads(output.replace('Product foo:  Found in channel test_channel\n', ''))
+    elements = list(as_json.keys())
+    elements.sort()
+
+    assert elements == ['channel', 'generation_id', 'key1', 'key2', 'taskmanager_id']
+    for element in elements:
+        assert isinstance(as_json[element], dict)
+
+    assert as_json['channel']['0'] == 'test_channel'
+    assert as_json['channel']['1'] == 'test_channel'
+    assert as_json['generation_id']['0'] == 1
+    assert as_json['generation_id']['1'] == 1
+    assert as_json['generation_id']['2'] == 1
+    assert as_json['generation_id']['3'] == 2
+    assert as_json['taskmanager_id']['0'] == 1
+    assert as_json['taskmanager_id']['1'] == 1
+    assert as_json['key1']['0'] == 'value1'
+    assert as_json['key2']['0'] == 0.1
+    assert as_json['key1']['1'] == 'value2'
+    assert as_json['key2']['1'] == 2
+    assert as_json['key1']['2'] == 'value3'
+    assert as_json['key2']['2'] == 'Test'
+    assert as_json['key1']['3'] == 'value1'
+    assert as_json['key2']['3'] == 0.1
```

Comment:
In the new comparison, you don't check the order anymore. I guess it is secondary, we don't care about checking the formatting as long as it is correct json.

Reply:
That was my thinking, the order of the keys is non-deterministic once parsed.